### PR TITLE
[Improve](nereids) use hash set replace three set in DiscreteValue  to improve in predicate performance

### PR DESCRIPTION
**Pull request by qzsee**
rewriting is slow if you use the in predicate in a query that contains more than 26,000 elements

Found by the flame image

image

tree sets sort slowly

There is no need to sort here, just use hashset

rewriting time spent from 3s to 1s

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java [hunk 1]
```diff
@@ -408,7 +408,7 @@ public DiscreteValue(ExpressionRewriteContext context,
         public DiscreteValue(ExpressionRewriteContext context,
                 Expression reference, Expression toExpr, Collection<Literal> values) {
             super(context, reference, toExpr);
-            this.values = Sets.newTreeSet(values);
+            this.values = Sets.newHashSet(values);
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java:411]: may cause options of `in` is no order.

check test on a in [1, 10, 20] union a in [100, 200] => a in [1, 10, 20, 100, 200]

**Comment by yujun777** [comment]
Literal.toLegacyLiteral is time cost.  can it possible to this function instead ? for example, use a lazy supply function

**Review by liutang123 — DISMISSED** [review]

**Comment by dqz123** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Comment by dqz123** [comment]
run buildall

**Comment by dqz123** [comment]
run performance

**Comment by dqz123** [comment]
run performance

**Comment by dqz123** [comment]
run buildall

**Review by liutang123 — APPROVED** [review]

**Comment by liutang123** [comment]
run buildall

**Comment by morrySnow** [comment]
run p0

**Review by cambyzju — APPROVED** [review]

**Pull request merged**